src/cdk-experimental/listbox/listbox.ts

```diff
@@ -194,360 +181,541 @@ export class CdkOption<T = unknown> implements ListKeyManagerOption, Highlightab
+
+  /** The child options in this listbox. */
+  @ContentChildren(CdkOption, {descendants: true}) protected options: QueryList<CdkOption<T>>;
+
+  /** The selection model used by the listbox. */
+  protected selectionModelSubject = new BehaviorSubject(
+    new SelectionModel<T>(this.multiple, [], true, this._compareWith),
+  );
 
-  /** Determines the orientation for the list key manager. Affects keyboard interaction. */
-  @Input('listboxOrientation') orientation: 'horizontal' | 'vertical' = 'vertical';
+  /** The key manager that manages keyboard navigation for this listbox. */
+  protected listKeyManager: ActiveDescendantKeyManager<CdkOption<T>>;
 
-  @Input() compareWith: (o1: T, o2: T) => boolean = (a1, a2) => a1 === a2;
+  /** Emits when the listbox is destroyed. */
+  protected readonly destroyed = new Subject<void>();
 
-  constructor(
-    @Optional() @Inject(CDK_COMBOBOX) private readonly _combobox: CdkCombobox,
-    @Optional() private readonly _dir?: Directionality,
-  ) {}
+  /** The host element of the listbox. */
+  protected readonly element: HTMLElement = inject(ElementRef).nativeElement;
 
-  ngOnInit() {
-    this._selectionModel = new SelectionModel<CdkOption<T>>(this.multiple);
+  /** The change detector for this listbox. */
+  protected readonly changeDetectorRef = inject(ChangeDetectorRef);
+
+  /** Callback called when the listbox has been touched */
+  private _onTouched: () => {};
+
+  /** Callback called when the listbox value changes */
+  private _onChange: (value: T[]) => void = () => {};
+
+  /** Callback called when the form validator changes. */
+  private _onValidatorChange = () => {};
+
+  /** Emits when an option has been clicked. */
+  private _optionClicked = defer(() =>
+    (this.options.changes as Observable<CdkOption<T>[]>).pipe(
+      startWith(this.options),
+      switchMap(options => merge(...options.map(option => option._clicked.pipe(mapTo(option))))),
+    ),
+  );
+
+  /** The directionality of the page. */
+  private readonly _dir = inject(Directionality, InjectFlags.Optional);
+
+  // TODO(mmalerba): Should not depend on combobox
+  private readonly _combobox = inject(CdkCombobox, InjectFlags.Optional);
+
+  /**
+   * Validator that produces an error if multiple values are selected in a single selection
+   * listbox.
+   * @param control The control to validate
+   * @return A validation error or null
+   */
+  private _validateMultipleValues: ValidatorFn = (control: AbstractControl) => {
+    const controlValue = this._coerceValue(control.value);
+    if (!this.multiple && controlValue.length > 1) {
+      return {'cdkListboxMultipleValues': true};
+    }
+    return null;
+  };
+
+  /**
+   * Validator that produces an error if any selected values are not valid options for this listbox.
+   * @param control The control to validate
+   * @return A validation error or null
+   */
+  private _validateInvalidValues: ValidatorFn = (control: AbstractControl) => {
```

Comment:
Should we always apply this validator or let the user optionally apply it?

Reply:
My thinking is that maybe someone would have a combobox where the user can either type freeform or or select an option from the listbox. I think in this case the user would probably bind the `formControl` to the `<input>` which would then communicate with the listbox? So maybe its ok to just always add this validator? WDYT?

Reply:
The only way to get into this situation would be programmatically, right? For an end-user, if you're only interacting with the rendered options, there wouldn't be a way to select a non-option value. 

If so, I wonder if it makes sense to treat this as an `ngDevMode` error instead of an end-user validation error.

Reply:
If you hook 2 different inputs to the same `FormControl` you can get in this situation. I actually added such a case to the dev app so I could play with it. Let me try to get a demo instance running for this PR (I think I have to make it non-draft for that to work?)

Reply:
https://ng-comp-dev--pr-24920-2aed840e65107482bee27fe38435bb56-7fidqw0e.web.app/cdk-experimental-listbox
- click "Kiwi" in the first listbox
- second listbox shows error

Reply:
@jelbourn are you satisfied with a forms error in this case?